plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -378,13 +378,13 @@ private static void validateCertificates(KeyStore keyStore)
     private Set<ElasticsearchNode> fetchNodes()
     {
         NodesResponse nodesResponse = doRequest("/_nodes/http", NODES_RESPONSE_CODEC::fromJson);
-
+        Set<String> nodeRoles = Set.of("data", "data_hot", "data_content");
         ImmutableSet.Builder<ElasticsearchNode> result = ImmutableSet.builder();
         for (Map.Entry<String, NodesResponse.Node> entry : nodesResponse.getNodes().entrySet()) {
             String nodeId = entry.getKey();
             NodesResponse.Node node = entry.getValue();
 
-            if (node.getRoles().contains("data")) {
+            if (node.getRoles().stream().anyMatch(nodeRoles::contains)) {
```

Comment:
node.getRoles() is a set and so we can reasonable expect that `.contains` is a constant time while `.stream().anyMatch()` be linear. i don't think it matters here (how many roles a node can have?), but as a general practice, we should use the proper APIs.
Maybe 
```
!Sets.intersection(node.getRoles(), DATA_ROLES).isEmpty()
```